awx/sso/pipeline.py

```diff
@@ -138,3 +143,73 @@ def update_user_teams(backend, details, user=None, *args, **kwargs):
         users_expr = team_opts.get('users', None)
         remove = bool(team_opts.get('remove', True))
         _update_m2m_from_expression(user, team.member_role.members, users_expr, remove)
+
+
+def update_user_orgs_by_saml_attr(backend, details, user=None, *args, **kwargs):
```

Comment:
There should be something here we can use to determine if the user actually authenticated via SAML vs. another social auth mechanism. If a SAML attribute is defined, but they log in another way, it will result in the user being removed from all orgs.

Reply:
@cchurch are you pointing out that in order to support the "ability to login via a social auth other than the one the user was imported from" feature https://github.com/ansible/awx/issues/308 that we would need to add something here. Or are you saying that in order for this feature to be complete, this change is needed?

Reply:
I'm saying it should be implemented in order for it to be complete. The pipeline will run for all configured social auth providers, but this function only needs to run when it can be determined that the user authenticated using SAML.